src/movie/Edit.js

```diff
@@ -0,0 +1,183 @@
+import React, { useReducer, useState } from 'react';
+import { useParams, useHistory } from 'react-router';
+import { Link } from 'react-router-dom';
+import { Alert } from 'react-bootstrap';
+import { movieList, genres } from './Const';
+
+import './Edit.scss';
+
+export function EditMovie() {
+  const [formErrorShow, setFormErrorShow] = useState(false);
+  const [submitSuccessShow, setSubmitSuccessShow] = useState(false);
+  const history = useHistory();
+
+  let { id } = useParams();
+  id = Number(id) || 0;
+  const isAddMovie = !id;
+  const movieDetails = isAddMovie ? {} : movieList.find((movie) => movie.id === id);
+
+  const formReducer = (movieFormState, newMovieFormState) => ({
+    ...movieFormState,
+    ...newMovieFormState,
+  });
+  const [movieFormState, dispatchInputChange] = useReducer(
+    formReducer,
+    movieDetails
+  );
+
+  const getMultiSelectValue = (element) => {
+    const selectedOptions = Array.prototype.slice.call(element.selectedOptions);
+    return selectedOptions.map((option) => option.value);
+  };
+
+  const handleInputChange = (event) => {
+    const value =
+      event.target.type === 'select-multiple' // switch to switch-case if condition increases
+        ? getMultiSelectValue(event.target)
+        : event.target.value;
+    dispatchInputChange({ [event.target.name]: value });
+  };
+
+  const resetForm = () => {
+    setFormErrorShow(false);
+    dispatchInputChange(movieDetails);
+  };
+
+  const validateForm = () => {
+    if (!movieFormState.title || !movieFormState.release_date) {
+      setFormErrorShow(true);
+      throw new Error('Missing form fields!');
+    }
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    validateForm();
+    setSubmitSuccessShow(true);
+    setTimeout(() => history.push('/'), 3000);
+  };
+
+  return (
+    <div className="router-page router-page-common edit-movie-page content-container text-white">
+      <Link to="/" className="btn header-cross-icon text-white">
+        &times;
+      </Link>
+      <h3 className="form-title">{isAddMovie ? 'ADD' : 'EDIT'} MOVIE</h3>
+      <form className="black-theme-form" autoComplete="off" onSubmit={handleSubmit}>
+        {!id ? (
+          ''
+        ) : (
+          <div className="form-group">
+            <label htmlFor="id">MOVIE ID</label>
+            <input
+              type="text"
+              className="form-control form-control-lg"
+              name="id"
+              defaultValue={movieFormState.id}
+            />
+          </div>
+        )}
```

Comment:
Actually I was using id=0 to represent `add` new movie scenario hence used ternary operator so that 0 wont show in ui. But  I will refactor it so that I can use && operator.